packages/components/src/utils/get-prop-attributes.ts

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright 2022 by Swiss Post, Information Technology
+ */
+
+import { toKebabCase } from './to-kebab-case';
+
+export function getPropAttributes(props: Record<string, unknown> | undefined): string {
+  if (!props) return '';
+
+  return Object.keys(props).reduce((attributes, propName) => {
+    const propValue = props[propName];
+    if (typeof propValue !== 'undefined') attributes += ` ${toKebabCase(propName)}="${propValue}"`;
+
+    return attributes;
+  }, ' ');
```

Comment:
It's just a different way to write your lines, please take over or leave out whatever you like.
But notice, that in your code you would write `propName="propValue"` even if the propValue is an empty string or null.

```suggestion
  return Object.entries(props)
    .reduce((attributes, [ propName, propValue ]) => attributes.concat(propValue ? ` ${toKebabCase(propName)}="${propValue}"` : '', ' ');
```

Reply:
I'm still used to avoid `Object.entries` because of IE but you're right, that time is over! I just updated this.

Regarding the solution you proposed, if the propValue is "0" or "false", it will not be rendered.
With Stencil components, we should only have props with a default value (different from null or an empty string) or no value at all (undefined). For me, the only cases where a prop may be set to null or an empty string is at runtime when the user is playing with the component. In this case, s.he should be able to write anything and incorrect attributes would be caught with errors.